Training24/Test.c

```diff
@@ -1,14 +1,193 @@
 // Test.c
-// Program on main branch.
+// Program on A4 branch.
+// 1. Palindrome is a word that reads the same forward or backward. Example: Civic, Madam, Bib, Nun, Level and so on. 
+//    Your program must take the word as input and display the result if it is a palindrome or not. 
+//    For complexity, you can also use phrases instead of words (E.g., Was it a car or a cat I saw? I did, did I? Don�t nod)
+// 2. Write a program that reverses a given integer (as defined in C) as explained below and check if it is a palindrome.
 // ------------------------------------------------------------------------------------------------
 #include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <malloc.h>
+#include <ctype.h> 
+#include <stdbool.h>
+#include <conio.h> // For getch()
 #include "Header.h"
 
-int main () {
-   printf ("%s\n", GetString ());
-   return 0;
+// ANSI escape codes for colors
+#define RESET   "\033[0m"
+#define GREEN   "\033[32m"
+#define RED     "\033[31m"
+#define YELLOW  "\033[33m"
+#define MAGENTA "\033[35m"
+#define SKYBLUE "\033[36m"
+
+/// <summary>Function to clear the console screen.</summary>
+void ClearScreen () {
+#ifdef _WIN32
+   system ("cls");           // Windows-specific command to clear the screen
+#endif
+}
+
+void FilterChar (char* buffer, const char* temp) {
+   int j = 0;
+   for (int i = 0; temp[i] != '\0'; i++) if (isalnum (temp[i])) buffer[j++] = tolower (temp[i]); // Convert to lowercase and add to buffer
+   buffer[j] = '\0';
+}
+
+/// <summary>Function to print test results for string.</summary>
+void FormatResult (int numTests, const char** inputs, const char** actualResults, const int* pass) {
+   printf (YELLOW "------------------- Test Results for Palindrome Check --------------------\n" RESET"| Test Case | Input                           | Output            |Result|\n"
+      "|-----------|---------------------------------|-------------------|------|\n");
+   for (int i = 0; i < numTests; i++)
+      printf ("| %-9d | %-31s | %-17s | %6s |\n", i + 1, inputs[i], actualResults[i] ? actualResults[i] : "NULL", pass[i] ? GREEN "PASS" RESET : RED "FAIL" RESET);
+   printf ("\n");
+}
+
+/// <summary>Function to print test results for reverse number.</summary>
+void FormatResult2 (int numTests, int* inputs, int* actualResults, int* pass) {
+   printf (YELLOW "--------------------- Test Results for Reverse Number --------------------\n" RESET"| Test Case | Input                           | Output            |Result|\n"
+      "|-----------|---------------------------------|-------------------|------|\n");
+   for (int i = 0; i < numTests; i++) printf ("| %-9d | %-31d | %-17d | %8s |\n", i + 1, inputs[i], actualResults[i], pass[i] ? GREEN "PASS" RESET : RED "FAIL" RESET);
+   printf ("\n");
+}
+
+/// <summary>Test cases for Palindrome function.</summary>
+void TestPalindrome () {
+   struct {
+      const char* input;
+      const char* expected;

```

Comment:
Property names should start with capital letters.
```suggestion
      const char* Input;
      const char* Expected;

```